### PR TITLE
[GPU] Enable validate after adding reorder

### PR DESCRIPTION
**Pull request by kelvinchoi-intel**
### Contents:
 - *Enable validate after adding reorder: After adding reorder, valid_output_layout is false and this could cause unnecessary output recalculation.*
 - *Activation ref kernel doesn't support in case input and output dim size is different, so it should be invalidated for the case.*

### Tickets:
 - *67962*

### CODE REVIEW

File: src/plugins/intel_gpu/tests/test_cases/fusings_gpu_test.cpp [hunk 1]
```diff
@@ -10320,17 +10330,15 @@ TEST_P(conv_fp32_reorder_bfyx_to_fsv32_conv_data_padding, have_data_padding) {
...
 
     implementation_desc conv_impl = { format::fs_b_yx_fsv32, "" };
+    bo_not_fused.set_option(build_option::force_implementations({ { "conv_output", conv_impl } }));
     bo_fused.set_option(build_option::force_implementations({ { "conv_output", conv_impl } }));
 
     execute(p);
 }
 INSTANTIATE_TEST_SUITE_P(fusings_gpu, conv_fp32_reorder_bfyx_to_fsv32_conv_data_padding, ::testing::ValuesIn(std::vector<bc_test_params>{
-    bc_test_params{ FSV32_CASE_CONV_FP32_1,  5, 6 }
+    bc_test_params{ FSV32_CASE_CONV_FP32_1,  3, 3 }
```
Comment (yeonbok) [inline src/plugins/intel_gpu/tests/test_cases/fusings_gpu_test.cpp:10343]: why are these changed?

File: src/plugins/intel_gpu/tests/test_cases/fusings_gpu_test.cpp [hunk 2]
```diff
@@ -10320,17 +10330,15 @@ TEST_P(conv_fp32_reorder_bfyx_to_fsv32_conv_data_padding, have_data_padding) {
         data("weights_dw", get_mem(dw_weights_layout, -127, 127)),
         convolution("conv_prim", "input", { "weights" }, p.groups, p.stride, p.pad, p.dilation),
         reorder("reorder_fsv32", "conv_prim", layout(data_types::f32, format::fs_b_yx_fsv32, dw_tensor, padding{ { 0, 0, 1, 1 }, 0 })),
-        convolution("conv_output", "reorder_fsv32", { "weights_dw" }, p.out_shape.feature[0], dw_stride, p.pad, p.dilation),
-        activation("activation", "conv_output", activation_func::abs),
-        activation("activation2", "conv_prim", activation_func::abs),
-        eltwise("add_bias", { "activation", "activation2" }, eltwise_mode::sum)
+        convolution("conv_output", "reorder_fsv32", { "weights_dw" }, p.out_shape.feature[0], dw_stride, p.pad, p.dilation)
```
Comment (yeonbok) [inline src/plugins/intel_gpu/tests/test_cases/fusings_gpu_test.cpp:10333]: Why did you reduced the testing network ? I think this is not proper approach..

File: src/plugins/intel_gpu/tests/test_cases/fusings_gpu_test.cpp (hunk 1 above)
Comment (kelvinchoi-intel) [reply src/plugins/intel_gpu/tests/test_cases/fusings_gpu_test.cpp]: it's fixed.

File: src/plugins/intel_gpu/tests/test_cases/fusings_gpu_test.cpp (hunk 2 above)
Comment (kelvinchoi-intel) [reply src/plugins/intel_gpu/tests/test_cases/fusings_gpu_test.cpp]: it's fixed.

File: src/plugins/intel_gpu/tests/fusions/convolution_fusion_test.cpp [hunk 3]
```diff
@@ -113,6 +113,39 @@ class ConvFusingTest : public BaseFusingTest<convolution_test_params> {
...
+    void execute(convolution_test_params& p) {
+        auto input_prim = get_mem(get_input_layout(p));
+        network network_not_fused(this->engine, this->topology_non_fused, bo_not_fused);
+        network network_fused(this->engine, this->topology_fused, bo_fused);
+        network_fused.set_input_data("input", input_prim);
+        network_not_fused.set_input_data("input", input_prim);
+
+        compare(network_not_fused, network_fused, p, true);
+        auto find_conv = [](primitive_info& p) -> bool {
+            if (p.original_id == "conv_prim")
```
Comment (yeonbok) [inline src/plugins/intel_gpu/tests/fusions/convolution_fusion_test.cpp:127]: in your tests below, no "conv_prim" primitive_id is used. Is this needed?

File: src/plugins/intel_gpu/tests/fusions/convolution_fusion_test.cpp [hunk 4]
```diff
@@ -2564,11 +2602,11 @@ TEST_P(conv_fp32_reorder_bfyx_to_fsv32_conv_basic, basic) {
     execute(p);
 }
 INSTANTIATE_TEST_SUITE_P(fusings_gpu, conv_fp32_reorder_bfyx_to_fsv32_conv_basic, ::testing::ValuesIn(std::vector<convolution_test_params>{
-    convolution_test_params{ FSV32_CASE_CONV_FP32_1,  3, 3 }
+    convolution_test_params{ FSV32_CASE_CONV_FP32_1,  3, 4 }
```
Comment (yeonbok) [inline src/plugins/intel_gpu/tests/fusions/convolution_fusion_test.cpp:2605]: Is reorder_out being fused to convolution in the above TC? if so.. the # of fused network should be 2?

File: src/plugins/intel_gpu/tests/fusions/convolution_fusion_test.cpp [hunk 5]
```diff
@@ -2564,11 +2602,11 @@ TEST_P(conv_fp32_reorder_bfyx_to_fsv32_conv_basic, basic) {
...
 }));
 
-
-class conv_fp32_reorder_bfyx_to_fsv32_conv_mean : public ConvFusingTest {};
+// 'reorder_fsv32' should NOT be optimized because it has 'mean'
```
Comment (yeonbok) [inline src/plugins/intel_gpu/tests/fusions/convolution_fusion_test.cpp:2608]: I'd like to suggest to mention in a ligher way, e.g.,  "'reorder_fsv32' is not being fused in the current impl, since it has 'mean'.

File: src/plugins/intel_gpu/tests/fusions/convolution_fusion_test.cpp [hunk 6]
```diff
@@ -2589,11 +2627,11 @@ TEST_P(conv_fp32_reorder_bfyx_to_fsv32_conv_mean, have_mean) {
     execute(p);
 }
 INSTANTIATE_TEST_SUITE_P(fusings_gpu, conv_fp32_reorder_bfyx_to_fsv32_conv_mean, ::testing::ValuesIn(std::vector<convolution_test_params>{
-    convolution_test_params{ FSV32_CASE_CONV_FP32_1,  3, 3 }
+    convolution_test_params{ FSV32_CASE_CONV_FP32_1,  4, 4 }
 }));
 
-
-class conv_fp32_reorder_bfyx_to_fsv32_conv_subtract : public ConvFusingTest {};
+// 'reorder_fsv32' should NOT be optimized because it has 'subtract'
```
Comment (yeonbok) [inline src/plugins/intel_gpu/tests/fusions/convolution_fusion_test.cpp:2633]: same as above. "should" sounds very strong.

File: src/plugins/intel_gpu/tests/fusions/convolution_fusion_test.cpp [hunk 7]
```diff
@@ -2626,12 +2663,12 @@ INSTANTIATE_TEST_SUITE_P(fusings_gpu, conv_fp32_reorder_bfyx_to_fsv32_conv_subtr
     convolution_test_params{ FSV32_CASE_CONV_FP32_1,  4, 4 }
 }));
 
-
-class conv_fp32_reorder_bfyx_to_fsv32_conv_fused_activation : public ConvFusingTest {};
+// 'reorder_fsv32' should NOT be fused because it has 'fused_activation'
```
Comment (yeonbok) [inline src/plugins/intel_gpu/tests/fusions/convolution_fusion_test.cpp:2666]: same as above

File: src/plugins/intel_gpu/tests/fusions/convolution_fusion_test.cpp [hunk 8]
```diff
@@ -2682,7 +2718,7 @@ TEST_P(conv_fp32_reorder_bfyx_to_fsv32_conv_data_padding, have_data_padding) {
     execute(p);
 }
 INSTANTIATE_TEST_SUITE_P(fusings_gpu, conv_fp32_reorder_bfyx_to_fsv32_conv_data_padding, ::testing::ValuesIn(std::vector<convolution_test_params>{
-    convolution_test_params{ FSV32_CASE_CONV_FP32_1,  5, 6 }
+    convolution_test_params{ FSV32_CASE_CONV_FP32_1,  5, 5 }
```
Comment (yeonbok) [inline src/plugins/intel_gpu/tests/fusions/convolution_fusion_test.cpp:2721]: Is it fused? ( it is mentioned that this "should be fused" above). If so, why the # of primtives are same b/w fused and unfused?

File: src/plugins/intel_gpu/tests/fusions/convolution_fusion_test.cpp (hunk 3 above)
Comment (kelvinchoi-intel) [reply src/plugins/intel_gpu/tests/fusions/convolution_fusion_test.cpp:127]: fixed (conv_out to conv_prim)

File: src/plugins/intel_gpu/tests/fusions/convolution_fusion_test.cpp (hunk 4 above)
Comment (kelvinchoi-intel) [reply src/plugins/intel_gpu/tests/fusions/convolution_fusion_test.cpp:2605]: reorder_out is not fused. only reorder_fsv32 is fused.

Comment (yeonbok) [reply src/plugins/intel_gpu/tests/fusions/convolution_fusion_test.cpp:2605]: If so, this test is not for "fused reorder which is converting fsv conv output layout to bfyx layout" ?

File: src/plugins/intel_gpu/tests/fusions/convolution_fusion_test.cpp (hunk 8 above)
Comment (kelvinchoi-intel) [reply src/plugins/intel_gpu/tests/fusions/convolution_fusion_test.cpp:2721]: 'reorder_fsv32' is fused in the point without 'enable_reorder_fusing'. As the result, both fused and not-fused are fused.

File: src/plugins/intel_gpu/tests/fusions/convolution_fusion_test.cpp (hunk 5 above)
Comment (kelvinchoi-intel) [reply src/plugins/intel_gpu/tests/fusions/convolution_fusion_test.cpp]: applied

File: src/plugins/intel_gpu/tests/fusions/convolution_fusion_test.cpp (hunk 6 above)
Comment (kelvinchoi-intel) [reply src/plugins/intel_gpu/tests/fusions/convolution_fusion_test.cpp]: applied

File: src/plugins/intel_gpu/tests/fusions/convolution_fusion_test.cpp (hunk 7 above)
Comment (kelvinchoi-intel) [reply src/plugins/intel_gpu/tests/fusions/convolution_fusion_test.cpp]: applied

File: src/plugins/intel_gpu/tests/fusions/convolution_fusion_test.cpp (hunk 4 above)
Comment (kelvinchoi-intel) [reply src/plugins/intel_gpu/tests/fusions/convolution_fusion_test.cpp:2595]: This test is for "fused reorder which is converting bfyx to fsv32, in front of convolution"

**Review by yeonbok — APPROVED** [review]
